lib/optimist-config-file.js

```diff
@@ -30,6 +31,9 @@ function parseConfigFile(configFileName) {
 	// parse it
 	try {
 		raw = fs.readFileSync(configFileName).toString();
+		if (replacementVars instanceof Object) {
+			raw = replace(raw, replacementVars);
+		}
```

Comment:
True.  Ok, updated the PR, and added a test to assert that variables aren't replaced in JSON, so at least the behavior is documented.